pommerman/network/ion_server/network.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
+"""IonServer Network handler - This contains functions responsible Server-Client communication (You shouldn't use this file directly)"""
+
+# Imports
+from asyncio import new_event_loop, set_event_loop, get_event_loop
+from websockets import serve
+from threading import Thread
+from .constants import SPCommands as spc
+from .constants import NetworkCommands as nic
+from time import time, sleep
+from os import getcwd, path
+from re import fullmatch
+from gzip import compress
+from rapidjson import loads, dumps, load
+from json import JSONDecodeError
+from uuid import uuid4
+
+# Global variables
+CONCURRENTLY_LOOKING = {"room": {}, "noroom": []}
+PLAYER_WS = {}
+MATCH_PROCESS = {}
+THREAD_LOCK = False
+
+# Functions
+async def decode_message(message, websocket):
+    """Decode the messages recieved from the clients"""
+    if message["i"] == nic.check.value:
+        await websocket.send(
+            dumps({
+                "i": nic.status_ok.value,
+                "p": len(PLAYER_WS),
+                "m": len(MATCH_PROCESS)
+            }))
+    elif message["i"] == nic.match_act.value:
+        if message["sid"] == MATCH_PROCESS[message["u"]]["id"]:
+            MATCH_PROCESS[message["u"]]["act"][MATCH_PROCESS[message["u"]][
+                "pl"].index(message["id"])] = message["act"]
+            MATCH_PROCESS[message["u"]]["recv"][MATCH_PROCESS[message["u"]][
+                "pl"].index(message["id"])] = True
+    elif message["i"] == nic.replay.value:
+        try:
+            with open(
+                    path.join(
+                        path.join(getcwd(), "matches"),
+                        str(message["p"]) + ".json"), 'r') as f:
+                # Note: Registry expression match comes after as it's an expensive operation as compared to file I/O
+                if fullmatch("^[a-z0-9-]*$", message["p"]) is not None:
+                    f = load(f)
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_ok.value, f]), "utf8")))
+                else:
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+        except:
+            await websocket.send(
+                compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+    elif len(PLAYER_WS) < MAX_PLAYERS and (message["i"] == nic.match.value
+                                           or message["i"] == nic.room.value):
+        uuid = str(uuid4())
+        while uuid in PLAYER_WS:
```

Comment:
What is this doing?

Reply:
There's another bug in there. You can try guessing what it was. Anyway, the UUID part generates a UUID for the user.